code/graph_algorithms/src/centroid_decomposition/centroid_decompostition.hpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include <vector>
+
+class Graph 
+{
+	std::vector<int> empt;
+	int ver_;
+	std::vector<std::vector<int>> adj_;
+	std::vector<std::vector<int>> centroidTree_;
+	std::vector<int> sizes_;
+	std::vector<bool> marked_;
+    
+	Graph(int num) :
```

Comment:
None of these functions are public, As such, no one outside your class will be able to access them. Do this:

```C++
    std::vector<bool> marked_;

public:
    // All of your functions, including constructor
```